.github/workflows/reusable-build-and-push.yml

```diff
@@ -0,0 +1,84 @@
+name: Build images and push to registry
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        required: false
+        type: string
+      sha:
+        required: true
+        type: string
+    secrets:
+      DOCKER_HUB_USERNAME:
+        required: true
+      DOCKER_HUB_PASSWORD:
+        required: true
+
+jobs:
+  build-and-push:
+    name: Build images and push
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
+        with:
+          ref: ${{ inputs.sha }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ vars.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Build and push frontend docker image
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          file: .docker-hub/frontend/Dockerfile
+          tags: |
```

Comment:
Previously, I had to separate multiple tags with a comma. Now you are separating them with a newline. Both are [supported](https://github.com/docker/build-push-action#inputs), but are you sure that the empty line on non-dev builds is tolerated by the build-push-action?

Reply:
It wasn't 100% clear from the documentation if this work, but tested and looks good.

### Test build on staging branch (additional tag)
https://github.com/usu/ecamp3/actions/runs/4460123937/jobs/7833062307
```
/usr/bin/docker buildx build --cache-from type=gha,scope=frontend --cache-to type=gha,scope=frontend,mode=max --file .docker-hub/frontend/Dockerfile --iidfile /tmp/docker-build-push-uul1Cu/iidfile --provenance mode=max,builder-id=https://github.com/usu/ecamp3/actions/runs/4460123937 --tag ursup/ecamp3-frontend:staging --tag ursup/ecamp3-frontend:92ee0671f1d2e2ca54ccd9f456a06333512d3490 --metadata-file /tmp/docker-build-push-uul1Cu/metadata-file --push .
```

### Test build on feature branch (no additional tag)
https://github.com/usu/ecamp3/actions/runs/4460124973/jobs/7833065137
```
/usr/bin/docker buildx build --cache-from type=gha,scope=frontend --cache-to type=gha,scope=frontend,mode=max --file .docker-hub/frontend/Dockerfile --iidfile /tmp/docker-build-push-BNSttV/iidfile --provenance mode=max,builder-id=https://github.com/usu/ecamp3/actions/runs/4460124973 --tag ursup/ecamp3-frontend:6fdcd39d2fbb022e453cf7209844720d391105cd --metadata-file /tmp/docker-build-push-BNSttV/metadata-file --push .
```